src/Controls/src/Core/Shapes/Shape.cs

```diff
@@ -261,5 +262,24 @@ PathF IShape.PathForBounds(Graphics.Rect viewBounds)
 
 			return path;
 		}
+
+		protected override Size MeasureOverride(double widthConstraint, double heightConstraint)
+		{
+			var result = base.MeasureOverride(widthConstraint, heightConstraint);
+			if (result.Width != 0 && result.Height != 0)
+			{
+				return result;
+			}
+
+			var boundsByFlattening = this.GetPath().Bounds.Size;
```

Comment:
In general everything looks really nice but there are a couple of wrong cases:

![image](https://user-images.githubusercontent.com/6755973/162424073-4acd03a5-d8f9-4431-94e1-f819eb4a2cf0.png)
This ellipse should fill all the yellow space.

![image](https://user-images.githubusercontent.com/6755973/162424163-757dac90-f396-46f5-9285-0b69defe44cb.png)
This should render a spiral path in the center.

Sometimes the bounds by flattening are wrong. For example, in the second case, Width and Height is zero.

Reply:
if the spiral path scenario is similar to one in repro project for #4044, then I believe the reason for the 0x0 size is that we do not invalidate the Height/Width request for the PolyLine when after replacing its Points. I think we should. 
If I do invalidation, then the spiral path renders correctly for me (at least in the repro project):

![image](https://user-images.githubusercontent.com/8218165/162536864-3b8b99cb-6768-4c3a-b11b-82ab423d7384.png)

Reply:
As for the ellipse scenario, I think my repro does not have an equivalent. There is a scenario with ellipses, but it is just a few simple fixed sized ellipse shapes and they show just fine. 

How can I get the AutoSize Shapes Gallery ?

Reply:
Here you can find the AutoSize sample https://github.com/dotnet/maui/blob/034167fdbf18e87e366f75e78e4bef076c488b16/src/Controls/samples/Controls.Sample/Pages/Controls/ShapesGalleries/AutoSizeShapesGallery.xaml

Reply:
Open the .NET MAUI Gallery and navigate to Controls -> Shapes. Then tap the "More samples" Button and select the "AutoSize Shapes Gallery".

Reply:
There were two issues:
- `GetBoundsByFlattening` with default `0.01` flatness produces clearly incorrect results for ellipse curves. Using `flatness == 1` temporarily is a workaround, but there is likely a bug in `GetBoundsByFlattening`
- `MeasureOverride` on Shape should consider Aspect. That was missing

Reply:
After the last changes:

![image](https://user-images.githubusercontent.com/8218165/164292812-26feb582-772a-4195-8d91-6637fca3cd4c.png)

![image](https://user-images.githubusercontent.com/8218165/164293041-a1a30053-ac3b-414b-957c-9d6876e141e7.png)